### PR TITLE
Kubernetes CNP status on invalid policies. 

### PR DESCRIPTION
**Pull request by eloycoto**
Read per commit: 

1) Fix for old Kubernetes versions, make sure that always return a ParseError.
2) Fix a timeout on Nightly, that makes thing a bit difficult to debug. 
3) Add a e2e test to validate that everything works as expected in all kubernetes versions.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7595)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-missed-k8s

File: pkg/k8s/cnp.go [hunk 1]
```diff
@@ -124,6 +124,7 @@ func (c *CNPStatusUpdateContext) prepareUpdate(cnp *types.SlimCNP, scopedLog *lo
 		serverRule = localCopy.DeepCopy()
 		_, err = serverRule.Parse()
 		if err != nil {
+			err = ErrParse{err.Error()}
```
Comment (ianvernon) [inline pkg/k8s/cnp.go:127]: What is the significance of this being a parse error vs. not being one?

Comment (eloycoto) [reply pkg/k8s/cnp.go:127]: If it's parse error it'll updated the status in the CNP if not, no.

**Review by ianvernon — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please (Hit https://github.com/cilium/cilium/issues/7598)

**Comment by eloycoto** [comment]
test-missed-k8s

File: test/k8sT/Policies.go [hunk 2]
```diff
@@ -256,6 +258,27 @@ var _ = Describe("K8sPolicyTest", func() {
...
 		}, 500)
 
+		It("Invalid Policy report status correctly", func() {
+			manifest := helpers.ManifestGet("invalid_cnp.yaml")
+			cnpName := "foo"
+			kubectl.Apply(manifest).ExpectSuccess("Cannot apply policy manifest")
+
+			body := func() bool {
+				cnp := kubectl.GetCNP(helpers.DefaultNamespace, cnpName)
+				if cnp != nil && len(cnp.Status.Nodes) > 0 {
```
Comment (nebril) [inline test/k8sT/Policies.go:268]: Can we check for the error here instead of only length of the status nodes?

**Review by nebril — CHANGES_REQUESTED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-missed-k8s

**Review by nebril — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Comment by vadorovsky** [comment]
Commit 5771397925a2c3861ef475dc7249ac7b11ebc071 definitely needs to be backported to 1.4 in order to fix the failure in Nightly CI:

https://jenkins.cilium.io/view/Cilium-v1.4/job/cilium-v1.4-Nightly/52/

But it probably makes sense to backport the whole pull request.

**Comment by vadorovsky** [comment]
^ The mentioned commit fixes the timeout from nanoseconds to seconds in models/policy helper.

**Pull request merged**